### PR TITLE
Add an additional method and test to calculate the projected distance along the major and minor axis (issue #1012)

### LINKED ISSUE #1012: Add option to project `separation` onto an ellipse
Right now separation only gives the angle between two points.  Another common "separation" in astronomy is the distance _projected_ along a particular axis (e.g. the major axis of a galaxy).  This would be particularly useful because of the rather sordid history of various people confusing which way a PA goes ("degrees east of north" I think is the most common convention).

Also useful (at least in galaxy contexts) might be an "elliptical" distance  (where the separation along the major and minor axes is weighted by the ellipticity).

This should probably await until some of the significant coordinate internal changes are done, so I'm not targeting a particular milestone.

### PR DESCRIPTION
**Pull request by datajungler**
### Description

This pull request is to create an additional method to derive the distance projected along the major and minor axis of the ellipse with known eccentricity. If I have any misunderstandings regarding this issue, please point them out and I will make corrections accordingly. The documentation shall be provided if the code works effectively and precisely resolves the issue.

Fixes #1012 

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/coordinates/baseframe.py [hunk 1]
```diff
@@ -1856,6 +1856,57 @@ def separation_3d(self, other):
...
+        sep_a : `~astropy.coordinates.Distance`
+            The real-space distance between these two coordinates along the major axis.
+        sep_b : `~astropy.coordinates.Distance`
+            The real-space distance between these two coordinates along the minor axis.
+
+        Raises
+        ------
+        ValueError
+            If this or the other coordinate do not have distances.
+            If the range of eccentricity is out of [0.0, 1.0]
```
Comment (pllim) [inline astropy/coordinates/baseframe.py:1882]:
```suggestion
            If the range of eccentricity is out of ``[0.0, 1.0]``.
```

**Comment by pllim** [comment]
Thanks for tackling a very old issue! Would need a change log as well.

Given we're less than a week from feature freeze, I am not sure if this will get in in time.

**Comment by eerovaher** [comment]
A general remark: having commits that fail the `pre-commit.ci` check means that the development environment has not been set up properly. `astropy` has [documentation describing how to use `pre-commit` locally.](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).

More specifically about this pull request, the function @eteq envisioned in #1012 would return a single `Angle`, not a pair of `Distance` instances, and in any case it must be possible for a user to specify the orientation of the ellipse on the sky.

**Comment by datajungler** [comment]
> A general remark: having commits that fail the `pre-commit.ci` check means that the development environment has not been set up properly. `astropy` has [documentation describing how to use `pre-commit` locally.](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
> 
> More specifically about this pull request, the function @eteq envisioned in #1012 would return a single `Angle`, not a pair of `Distance` instances, and in any case it must be possible for a user to specify the orientation of the ellipse on the sky.

Thank you for clarification. So, it is required to add an extra argument of "ellipse_position_angle", i.e. orientation. The output should be the angular separation between two points along this orientation. Please correct me if my understanding is wrong. 

But, how about the separation along the minor axis? Apologized that I might be confused to visualize the definition of this "elliptical" separation.

**Comment by eerovaher** [comment]
I am not very familiar with extragalactic astronomy, so I can't tell what conventions are being used by that community. Maybe @eteq as the author of the issue can clarify or provide a reference.

**Pull request closed without merging**